databuilder/extractor/dashboard/tableau/tableau_dashboard_extractor.py

```diff
@@ -0,0 +1,139 @@
+    DATABASE = const.DATABASE
+
+    def init(self, conf):
+        # type: (ConfigTree) -> None
+
+        self._conf = conf
+        self._auth = TableauDashboardAuth(self._conf)
+        self.query = """query {
+            workbooks {
+                id
+                name
+                createdAt
+                description
+                projectName
+                projectVizportalUrlId
+                vizportalUrlId
+            }
+        }"""
+
+        self._extractor = self._build_extractor()
+
+        transformers = []
+        timestamp_str_to_epoch_transformer = TimestampStringToEpoch()
+        timestamp_str_to_epoch_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf, timestamp_str_to_epoch_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict({FIELD_NAME: 'created_timestamp', })))
+        transformers.append(timestamp_str_to_epoch_transformer)
+
+        dict_to_model_transformer = DictToModel()
+        dict_to_model_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf, dict_to_model_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict(
+                    {MODEL_CLASS: 'databuilder.models.dashboard.dashboard_metadata.DashboardMetadata'})))
+        transformers.append(dict_to_model_transformer)
+        self._transformer = ChainedTransformer(transformers=transformers)
+
+    def extract(self):
+        # type: () -> Any
+
+        record = self._extractor.extract()
+        if not record:
+            return None
+
+        return self._transformer.transform(record=record)
+
+    def get_scope(self):
+        # type: () -> str
+
+        return 'extractor.tableau_dashboard_metadata'
+
+    def _build_extractor(self):
+        """
+        Builds a TableauGraphQLApiMetadataExtractor. All data required can be retrieved with a single GraphQL call.
+        :return: A TableauGraphQLApiMetadataExtractor that provides core dashboard metadata.
+        """
+        # type: () -> TableauGraphQLApiMetadataExtractor
+        extractor = TableauGraphQLApiMetadataExtractor()
+        tableau_extractor_conf = \
+            Scoped.get_scoped_conf(self._conf, extractor.get_scope())\
+                  .with_fallback(self._conf)\
+                  .with_fallback(ConfigFactory.from_dict({STATIC_RECORD_DICT: {'product': 'tableau'}
+                                                          }
+                                                         )
+                                 )
+        extractor.init(conf=tableau_extractor_conf, auth_token=self._auth.token, query=self.query)
+        return extractor
+
+
+class TableauGraphQLApiMetadataExtractor(TableauGraphQLApiExtractor):
+    """
+    Implements the extraction-time logic for parsing the GraphQL result and transforming into a dict
+    that fills the DashboardMetadata model. Allows workbooks to be exlcuded based on their project.
+    """
+    def execute(self):
```

Comment:
Add the config interface values here
```python
    TABLEAU_HOST = const.TABLEAU_HOST
    EXCLUDED_PROJECTS = const.EXCLUDED_PROJECTS
    CLUSTER = const.CLUSTER
```

Reply:
I was going back and forth on doing this, since they're implemented in the superclass and aren't different in the subclass I didn't copy them here. This is also why I was using the generic `TableauGraphQLApiExtractor` instead of `TableauGraphQLApiMetadataExtractor`, but if I were to copy them over then it would make sense to use `TableauGraphQLApiMetadataExtractor` instead. How do you feel about keeping the constants defined at `TableauGraphQLApiExtractor` and using `super` instead of directly calling `TableauGraphQLApiExtractor.CONSTANT`?